### PR TITLE
noisetorch: 0.12.0 -> 0.12.2

### PR DESCRIPTION
**Pull request by denbeigh2000**
###### Description of changes

Updates noisetorch to latest v0.12.2.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by Shawn8901 — APPROVED** [review]
lgfm, did some minor testing on my machine and dont see any obvious problems with the new version

**Comment by legendofmiracles** [comment]
```
Couldn't fetch latest release Get "https://api.github.com/repos/noisetorch/NoiseTorch/releases/latest": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
```
Can we disable the update mechanism? We couldn't use it either way.

**Comment by Shawn8901** [comment]
> ```
> Couldn't fetch latest release Get "https://api.github.com/repos/noisetorch/NoiseTorch/releases/latest": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
> ```
> 
> Can we disable the update mechanism? We couldn't use it either way.

i have briefly looked at the upstream project.
at some  point they seem to have a check if the update url is not set https://github.com/noisetorch/NoiseTorch/blob/0f1765c1e8bd5164fe895ae4aa13e3385740d0c0/Makefile#L1 && https://github.com/noisetorch/NoiseTorch/blob/0f1765c1e8bd5164fe895ae4aa13e3385740d0c0/update.go#L83

but on the other hand its hard coded.

https://github.com/noisetorch/NoiseTorch/blob/0f1765c1e8bd5164fe895ae4aa13e3385740d0c0/update.go#L162

some how looks like a bug for me (as mixing does not make so much sense and hard coding isnt a good idea).

someone could raise an issue upstream and kindly ask if they can add a way to disable automated updates, as its not very package manager friendly (even when not looking at nixpkgs).

**Comment by denbeigh2000** [comment]
> ```
> Couldn't fetch latest release Get "https://api.github.com/repos/noisetorch/NoiseTorch/releases/latest": context deadline exceeded (Client.Timeout exceeded while awaiting headers)
> ```
> 
> Can we disable the update mechanism? We couldn't use it either way.

https://github.com/noisetorch/NoiseTorch/blob/f204182782995ed01753b765fec861ee857e9220/main.go#L69-L71

We could, by potentially overriding `XDG_CONFIG_DIR` with a wrapper script and pointing it to a generated config in the nix store. Feels like a step too far, though, because that'd mean nobody could use custom configuration.

I'm personally just going to put a `.config/noisetorch/config.toml` on my systems with updates disabled.

**Comment by legendofmiracles** [comment]
I've created an issue: https://github.com/noisetorch/NoiseTorch/issues/345

In the meantime, I'm not sure what noisetorch will try to do when this gets out of date.
I think it's better to wait until the problem is sorted out than shipping in nixpkgs something that could soon be unstable.
Looking at the changelog it doesn't seem like there's any big change users would want as soon as possible

**Comment by denbeigh2000** [comment]
Update mechanism should be properly disabled once https://github.com/noisetorch/NoiseTorch/pull/355 is merged.

**Comment by TheDukeofErl** [comment]
On a somewhat related note, does Nix normally always use a versioned release? I ask because I don't think 0.12.3 (or whatever version will have the fix in) will be made for a little while longer: I've found some other issues with the update mechanism that will need to be fixed before that's done, amongst other things.

If Nix can use the git version (kind of like how the AUR handles things, with a "release" and "git" version), rather than the 0.12.2 version, that would let you guys have a good build sooner.

**Comment by denbeigh2000** [comment]
> On a somewhat related note, does Nix normally always use a versioned release? I ask because I don't think 0.12.3 (or whatever version will have the fix in) will be made for a little while longer: I've found some other issues with the update mechanism that will need to be fixed before that's done, amongst other things.
> 
> If Nix can use the git version (kind of like how the AUR handles things, with a "release" and "git" version), rather than the 0.12.2 version, that would let you guys have a good build sooner.

Thanks for the heads up! Would be up to @legendofmiracles as a maintainer, but I would imagine we'd probably either stay on 0.12.0, or build 0.12.2 and patch my commit on top.

**Comment by TheDukeofErl** [comment]
> or build 0.12.2 and patch my commit on top

I'd probably recommend against that for the time being... to keep it brief, 0.12.2 and releases prior to it have license issues with doing that :/

Going forward, applying patches shouldn't be an issue, but for the time being, I'd be very cautious and definitely advise against it. It's probably best to either go off of git, apply all the changes available in a patch (every commit from 0.12.2 to now), or just wait until it's sorted out (next release).

Sorry for any confusion on this front, I just really don't want to see Nix end up in a position like the AUR once was in. I can go into greater detail but I don't want to clog up the Nix repos with silly issues from upstream.

**Comment by Shawn8901** [comment]
I have played a bit around with the mentioned commit, from upstream. Looks fine for me so far.
Also the values seems to default back to empty, when not given on compile, which is also nice, as we dont have to change things there.
So we just have to decide if we want to wait for a next release.
In case someone wants to already use that, i have that in my flake

```
[...]
  pname = "NoiseTorch";
  version = "0.12.2.pre+date=2022-08-11";

  src = fetchFromGitHub {
    owner = "noisetorch";
    repo = pname;
    rev = "2792a642b8c483475d78518a0c148e5f68eb980b";
    sha256 = "sha256-mtGTa3jH7QR9RU6Gy9An7EX+B7ZM/+9EzlhBPxeQ/4M=";
    fetchSubmodules = true;
  };
[...]
```

**Review by SuperSandro2000 — COMMENTED** [review]
What is knownVulnerabilities and can it be removed? -s -w is missing from ldflags.

**Comment by Shawn8901** [comment]
> What is knownVulnerabilities and can it be removed?

I am not sure if the question goes to my direction. In the `meta` part of the derivation there is a field `knownVulnerabilities`. There was a security issue with the original project and the original author stopped the project. With 0.12 the project was forked/transfered to motivated community, who did then a code review to check if untrusty code was submitted.
0.12 is the first "trusted" again. Our derivation currently checks if the version is lower than (similar thing like for broken packages).

About breaking my snippet: 
As the `knownVulnerabilities` code block uses `lib.versionOlder` a version like `unstable-<date>` will cause `lib.versionOlder` to be true, which then results in printing that the version is affected by a known Vulnerability, which is not the case. So using another suggested version format from RFC107, is possibly "better" when not removing the `knownVulnerabilities` attribute. 

From my point of view enough time has passed to remove that warning again on master, but i am not the/a maintainer, but just a random dude using noisetorch, who has an interest in having a recent version.
But like i said i am not sure if you actually asked for that explanation.

> -s -w is missing from ldflags.

Thats actually strange, at least for me github does not show that as a difference, to there you are right, that should be added again, as its saving some MB of space.

**Comment by SuperSandro2000** [comment]
> 0.12 is the first "trusted" again. Our derivation currently checks if the version is lower than (similar thing like for broken packages).

So we can remove it because it is now always version 0.12.0 or above.

**Comment by SuperSandro2000** [comment]
FYI https://github.com/NixOS/nixpkgs/pull/217072

**Pull request closed without merging**
